lib/assertions.js

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright (c) 2021-2022 Digital Bazaar, Inc. All rights reserved.
+ */
+export function assertHttpsUrl(url) {
+  const _url = assertUrl(url);
+  assertHttps(_url);
+}
+
+export function assertUrl(url) {
+  if(url instanceof URL) {
+    return url;
+  }
+  if(typeof url === 'string') {
+    return new URL(url);
+  }
+  throw new TypeError('"url" must be a string or a url');
+}
+
+export function assertHttps(url) {
+  if(url.protocol === 'https:') {
+    return true;
+  }
+  throw new TypeError(
+    `"url" protocol must by "https:" received ${url.protocol}`);
+}
+
+export function assertDidWeb(did) {
+  const didType = typeof did;
+  if(didType !== 'string') {
+    throw new TypeError(`Expected did to be a string received ${didType}`);
+  }
+  const [scheme, method, domain] = did.split(':');
+  if(scheme !== 'did') {
+    const e = new Error(`Scheme must be "did" received ${scheme}`);
+    e.code = 'invalidDid';
+    throw e;
+  }
+  if(method !== 'web') {
+    const e = new Error(`Did method must be "web" received ${method}`);
```

Comment:
The other errors don't mention "DID".  Should be consistent one way or another.
```suggestion
    const e = new Error(`Method must be "web" received ${method}`);
```

Reply:
I have to disagree on this one. Having did with method removes ambiguity and make the error more readable. We don't want someone confused on what is meant with Method. It could easily be confused with numerous other methods in larger applications. I could capitalize did though.